scripts/west_commands/build.py

```diff
@@ -227,9 +230,24 @@ def do_run(self, args, remainder):
                     self.run_cmake = True
         else:
             self.run_cmake = True
+
         self.source_dir = self._find_source_dir()
         self._sanity_check()
 
+        build_info_path = os.path.join(self.build_dir, 'zephyr')
+        build_info_file = os.path.join(build_info_path, BUILD_INFO_LOG)
+        west_workspace = west_topdir(self.source_dir)
+        if not os.path.exists(build_info_path):
+            os.makedirs(build_info_path)
+        if not os.path.exists(build_info_file):
+            build_command = {'west': {'command': ' '.join(sys.argv[:]),
+                                     'topdir': str(west_workspace)}}
```

Comment:
Add the `west` version too.

Reply:
good point

Reply:
I guess this is true for almost all entries?

Version for cmake/python/...

Reply:
yep, providing the version of each tool we describe in the build info file is good.

Reply:
added `west` version in this PR.

Support for a list of tools and their versions will follow in a new PR.
Reason is I want to provide those as:
```
  tools:
    - name: python
      version: 3.10.4
    - name: CMake
      version: 3.24.5
```

however that requires that we guarantee this limitation is fixed:
https://github.com/zephyrproject-rtos/zephyr/blob/df7f1ff5b5b5cd18c68ccb676ca210b82b9fe6d4/cmake/modules/yaml.cmake#L32-L38